.travis.yml

```diff
@@ -74,136 +90,185 @@ matrix:
                CONDA_DEPENDENCIES='scipy'
                PYTEST_VERSION="<3.6.2"
 
+        # JOB .7
         # -> Have non Python package (openjpeg) in dependency list when
-        #    falling back on pip installing the dependencies (as photutils is
-        #    missing from listed channels) to test the install one-by-one.
+        #    falling back on pip installing the dependencies (as pyqt5 is
+        #    missing from listed channels, it is called pyqt) to test the install one-by-one.
         # -> Have package dependency that matches partially another package
         #    (pytest vs pytest-cov) when falling back to pip install
         # -> testing whether version numbers are respected by pip install fallback
-        - os: linux
-          env: SETUP_CMD='test --coverage' PYTHON_VERSION=3.4
+        # The following versions were updated in 2019 because the old versions
+        # were leading to issues because anaconda deprecated the old free channel:
+        #
+        #    PYTHON_VERSION 3.4 → 3.6
+        #    NUMPY_VERSION 1.10 → 1.16
```

Comment:
So this one is intended to check whether version numbers are respected by pip when the pip fallback is used? Or is the idea that we want the conda solve to fail for sure by using the combination of old numpy and newer scikit-learn that conda can't solve?

Reply:
yes, early on there was a bug that the pip install fallback didn't take into account the version pinning, e.g. `SCIKIT_LEARN_VERSION`. So this job needs a combination that has a package with a version that is not conda installable, but pip needs to respect the version being set in the env variable.

Reply:
ok; somewhat unrelated question: is it ok to write a pytthon script that is used by `setup_dependencies_common.sh`? I need a way to hack around a conda bug and it would be much easier to do the processing in python than in bash...

Reply:
yes, I suppose that's OK. We started this as a bash script as it was a simple ~50 line thing, then it started to grow organically to a monster. (Also I kind of like the bash hacking, but now parts of this are so much out of control that bugs remain undetected for quite a while).